src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidationsTest.java

```diff
@@ -258,6 +261,79 @@ void noCaseIdTest() {
         );
     }
 
+    @Test
+    void caseIdTest() {
+        checkValidation(
+            createCaseWith(b -> b.id(1L)),
+            true,
+            1L,
+            CallbackValidations::hasAnId,
+            null
+        );
+    }
+
+    @Test
+    void caseIdNoCaseDetailsTest() {
+        checkValidation(
+            null,
+            false,
+            1L,
+            CallbackValidations::hasAnId,
+            "Exception case has no Id"
+        );
+    }
+
+    @Test
+    void hasInvalidDataTest() {
+        CaseDetails caseDetails = createCaseWith(
+            b -> b
+                .id(null)
+                .caseTypeId("SERVICE_ExceptionRecord")
+                .jurisdiction("BULKSCAN")
+        );
+        Validation<Seq<String>, CaseDetails> res =
+            CallbackValidations.hasValidData(true, caseDetails);
+        assertThat(res.isValid()).isEqualTo(false);
+    }
+
+    @Test
+    void hasValidData_use_search_case_reference_true() {
```

Comment:
Regarding `useSearchCaseReference`, I think we need the following four scenarios covered:
- `useSearchCaseReference` == `true` and `searchCaseReference` is present => valid
- `useSearchCaseReference` == `true` and `searchCaseReference` is missing => invalid
- `useSearchCaseReference` == `false` and `attachToCaseReference` is present => valid
- `useSearchCaseReference` == `false` and `attachToCaseReference` is missing => invalid

Reply:
Done